### PR TITLE
homework_1

### PR DESCRIPTION
**Pull request by alyatertyshnikova**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -4,9 +4,99 @@
...
+            <artifactId>testng</artifactId>
+            <version>${testng.version}</version>
+            <scope>test</scope>
+        </dependency>
+
+        <dependency>
+            <groupId>com.epam.tat.module4</groupId>
+            <artifactId>calculator</artifactId>
+            <version>1.0</version>
+            <scope>compile</scope>
```
Comment (DmitryKhodakovsky) [inline pom.xml]: Why do you set `scope` **compile** for this `dependency`?

File: pom.xml [hunk 2]
```diff
@@ -4,9 +4,99 @@
...
+        </dependency>
+
+        <dependency>
+            <groupId>org.decimal4j</groupId>
+            <artifactId>decimal4j</artifactId>
+            <version>1.0.3</version>
+        </dependency>
```
Comment (DmitryKhodakovsky) [inline pom.xml]: Why do you need this dependency?

File: src/test/java/hw1/CalculatorMultiplyTest.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+    @Test(dataProvider = "multiplyDigitsDataProvider", groups = {"multiplyAndDivide"})
+    public void multiplyDigits(double a, double b, double expected) {
+        assertEquals(calculator.mult(a, b), expected);
+    }
+
+    @AfterMethod
+    public void tearDown() {
+        calculator = null;
+    }
```
Comment (DmitryKhodakovsky) [inline src/test/java/hw1/CalculatorMultiplyTest.java]: I could see that do you have code duplication in the following classes:
* CalculatorMultiplyTest
* CalculatorPowerOfDigitTest
* CalculatorSineOfDigitTest
* CalculatorSubtractionTest
* CalculatorIsNegativeTest
* CalculatorDivideTest

You have repeated field and methods in the each class.

File: src/test/resources/hw1/CalculatorTestSuite.xml [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+    <test name="Calculator test">
+        <classes>
+            <class name="hw1.CalculatorSubtractionTest"></class>
+            <class name="hw1.CalculatorMultiplyTest"></class>
+            <class name="hw1.CalculatorPowerOfDigitTest"></class>
+            <class name="hw1.CalculatorSineOfDigitTest"></class>
+            <class name="hw1.CalculatorIsNegativeTest"></class>
+            <class name="hw1.CalculatorDivideTest"></class>
+        </classes>
+    </test>
```
Comment (DmitryKhodakovsky) [inline src/test/resources/hw1/CalculatorTestSuite.xml]: Why do you decide include all list of classes in this suite? Do we have other way to create main suite for all tests?

**Review by DmitryKhodakovsky — CHANGES_REQUESTED** [review]
Please fix comments.
When I try to run test on my laptop with command `mvn clean test` than I've got an follow error:
```
EPRUPETW2288:AlinaTertyshnikova dmitry_khodakovsky$ mvn clean test
[INFO] Scanning for projects...
[INFO] 
[INFO] --------------------< com.epam:AlinaTertyshnikova >---------------------
[INFO] Building AlinaTertyshnikova 1.0-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[WARNING] The POM for com.epam.tat.module4:calculator:jar:1.0 is missing, no dependency information available
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  0.215 s
[INFO] Finished at: 2020-03-27T23:30:20+03:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal on project AlinaTertyshnikova: Could not resolve dependencies for project com.epam:AlinaTertyshnikova:jar:1.0-SNAPSHOT: Failure to find com.epam.tat.module4:calculator:jar:1.0 in https://repo.maven.apache.org/maven2 was cached in the local repository, resolution will not be reattempted until the update interval of central has elapsed or updates are forced -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/DependencyResolutionException
```

File: pom.xml (hunk 2 above)
Comment (alyatertyshnikova) [reply pom.xml]: I used it to round the result of sin method, but now I decide to do it in another way

**Review by DmitryKhodakovsky — APPROVED** [review]
HW1 is accepted. Could be merged into the master branch

**Pull request merged**